R/summaries.R

```diff
@@ -87,3 +105,231 @@ setMethod("getTopTaxa", signature = c(x = "SummarizedExperiment"),
+
+setGeneric("getUniqueTaxa",
+           signature = c("x"),
+           function(x, ...)
+             standardGeneric("getUniqueTaxa")
+)
+
+#' @rdname summaries-basic
+#' @export
+setMethod("getUniqueTaxa",
+          signature = c(x = "SummarizedExperiment"),
+          function(x, rank = NULL){
+            .check_taxonomic_rank(rank, x)
+            unique(rowData(x)[,rank])
+          }
+)
+
+
+#' @rdname summaries-basic
+#'
+#' @param rank A single character defining a taxonomic rank. Must be a value of
+#'   the output of \code{taxonomicRanks()}.
+#'
+#' @param group With group, it is possible to group the observations in an
+#'   overview. Must be a one of the column names of \code{colData}.
+#'
+#' @param name A name for the column of tibble table that includes taxa.
+#'
+#' @details
+#' \code{summarizeDominantTaxa} returns information about most dominant
+#' taxa in a tibble. Information includes their absolute and relative abundances in whole
+#' data set.
+#' \itemize{
+#'         \item{\code{rank}}{ with \code{rank} parameter, it is possible to
+#'         agglomerate taxa based on taxonomic ranks. E.g. if 'family' rank is
+#'         used, all abundances of same family is added together, and those
+#'         families are returned determining the threshold for coverage index.
+#'         By default, \code{threshold} is 0.9.}
+#'         \item{\code{group}}{ with \code{group} parameter, it is possible to
+#'         group observations of returned overview based on samples' features.
+#'         E.g., if samples contain information about patients' health status,
+#'         it is possible to group observations, e.g. to 'healthy' and 'sick',
+#'         and get the most dominant taxa of different health status.}
+#' }
+#'
+#'
+#' @return
+#' The \code{summarizeDominantTaxa} returns an overview in a tibble. It contains dominant taxa
+#' in a column named \code{*name*} and its abundance in the data set.
+#'
+#' @export
+setGeneric("summarizeDominantTaxa",signature = c("x"),
+           function(x,
+                    abund_values = "counts",
+                    rank = NULL,
+                    group = NULL,
+                    name = "dominant_taxa")
+               standardGeneric("summarizeDominantTaxa"))
+
+#' @rdname summaries-basic
+#' @export
+setMethod("summarizeDominantTaxa", signature = c(x = "SummarizedExperiment"),
+    function(x,
+             abund_values = "counts",
+             rank = NULL,
+             group = NULL,
+             name = "dominant_taxa"){
+        # Input check
+        # group check
+        if(!is.null(group)){
+            if(isFALSE(any(group %in% colnames(colData(x))))){
+                stop("'group' variable must be in colnames(colData(x))")
+                }
+            }
+        # Adds dominant taxas to colData
+        tmp <- dominantTaxa(x, abund_values, rank, name)
+        # Gets an overview
+        overview <- .get_overview(tmp, group, name)
+        overview
+  }
+)
```

Comment:
This function can basically summarize anything, which is a `character`/`factor`.

If `name = "whatever"` is set, the function no longer summarizes the dominant taxa. It summarizes whatever. From this I get the feeling that the name of the function and intended purpose of this function still needs to be discovered.

From the results of the function, I think this is a fancy version of `table` or specific version `tally`. Is that correct?

From the man page of `tally`, I got the idea, that a better function name for this function might actually by `countTaxa`. What are your thought on this?